apache/superset
# fix: cache warmup solution non legacy charts.

### SUMMARY
**Issue:** Cache-warmup API does not work as expected for all charts. Running cache-warmup for a slice ex: (vizType-Table) and exploring the same slice on browser gives different cache key.

**Solution:** /warm_up_cache/ API in current solution only applicable for legacy chart and those are very few in numbers, and when the same code executes for another type of chart such as Table chart and other non-legacy chart, it generates a different cache key not matching with the one we get while exploring the same slice on browser. Fix given here will check the viz type of the chart and handle it in the required way to update the right cache-key.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS
1. Create a dashboard that includes table chart.
2. Configure cache warm up schedule in superset_config.py.
3. Verify if the cache got refreshed after cache warm up execution using menu option at top-right corner of the chart.

### ADDITIONAL INFORMATION


- [x] Has associated issue: Fixes #15497
